pyincore/analyses/pipelinerestoration/pipelinerestoration.py

```diff
@@ -0,0 +1,200 @@
+# Copyright (c) 2019 University of Illinois and others. All rights reserved.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Mozilla Public License v2.0 which accompanies this distribution,
+# and is available at https://www.mozilla.org/en-US/MPL/2.0/
+
+
+import collections
+import concurrent.futures
+import pandas as pd
+
+from pyincore import BaseAnalysis, AnalysisUtil, RestorationService
+
+
+class PipelineRestoration(BaseAnalysis):
+    """
+    Args:
+        incore_client (IncoreClient): Service authentication.
+
+    """
+
+    def __init__(self, incore_client):
+        self.restorationsvc = RestorationService(incore_client)
+        super(PipelineRestoration, self).__init__(incore_client)
+
+    def get_spec(self):
+        """Get specifications of the Pipeline Restoration analysis.
+
+        Returns:
+            obj: A JSON object of specifications of the pipeline restoration analysis.
+
+        """
+        return {
+            'name': 'pipeline-restoration',
+            'description': 'calculate the restoration times for damaged pipelines',
+            'input_parameters': [
+                {
+                    'id': 'result_name',
+                    'required': True,
+                    'description': 'name of the result csv dataset',
+                    'type': str
+                },
+                {
+                    'id': 'num_cpu',
+                    'required': False,
+                    'description': 'If using parallel execution, the number of cpus to request',
+                    'type': int
+                },
+                {
+                    'id': 'num_available_workers',
+                    'required': True,
+                    'description': 'Number of available workers to work on the repairs',
+                    'type': int
+                },
+                {
+                    'id': 'restoration_key',
+                    'required': False,
+                    'description': 'restoration key to use in mapping dataset',
+                    'type': str
+                },
+            ],
+            'input_datasets': [
+                {
+                    'id': 'pipeline',
+                    'required': True,
+                    'description': 'Pipeline Inventory',
+                    'type': ['ergo:buriedPipelineTopology', 'ergo:pipeline'],
+                },
+                {
+                    'id': 'pipeline_damage',
+                    'required': True,
+                    'description': 'pipeline damage results with repairs',
+                    'type': ['ergo:pipelineDamageVer2']
```

Comment:
This should be ergo:pipelineDamageVer3 to match the output type from pipelinedamagerepairrate.py